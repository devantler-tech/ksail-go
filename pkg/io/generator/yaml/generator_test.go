package yamlgenerator_test

import (
	"testing"

	"github.com/devantler-tech/ksail-go/internal/testutils"
	generator "github.com/devantler-tech/ksail-go/pkg/io/generator/yaml"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestMain(m *testing.M) { testutils.RunTestMainWithSnapshotCleanup(m) }

func TestGenerate(t *testing.T) {
	t.Parallel()

	gen := generator.NewYAMLGenerator[map[string]interface{}]()
	data := map[string]interface{}{"name": "minimal"}
	result, err := gen.Generate(data, generator.Options{})
	require.NoError(t, err)
	require.NotEmpty(t, result)
	snaps.MatchSnapshot(t, result)
}
