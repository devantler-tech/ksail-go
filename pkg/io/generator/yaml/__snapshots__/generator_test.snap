
[TestGenerate - 1]
name: minimal

---
